package.json

```
@@ -1,91 +1,36 @@
 {
-  "name": "@metamask/controllers",
-  "version": "29.0.1",
+  "name": "@metamask/controllers-monorepo",
+  "version": "0.0.0",
+  "private": true,
   "description": "Collection of platform-agnostic modules for creating secure data models for cryptocurrency wallets",
-  "keywords": [
-    "MetaMask",
-    "Ethereum"
-  ],
-  "homepage": "https://github.com/MetaMask/controllers#readme",
-  "bugs": {
-    "url": "https://github.com/MetaMask/controllers/issues"
-  },
-  "repository": {
-    "type": "git",
-    "url": "https://github.com/MetaMask/controllers.git"
-  },
-  "license": "MIT",
-  "main": "./dist/index.js",
-  "types": "./dist/index.d.ts",
-  "files": [
-    "dist/"
+  "workspaces": [
+    "packages/*"
   ],
   "scripts": {
-    "build": "rimraf dist && tsc --project tsconfig.build.json",
-    "build:link": "yarn build && cd dist && yarn link && rm -rf node_modules && cd ..",
+    "// 1": "'pre' commands are not supported by Yarn >= 2",
+    "// prepublish": "yarn build",
+    "build": "tsc --build tsconfig.build.json --verbose",
+    "build:clean": "./scripts/clean-build.sh",
```

TypeScript will now generate `.tsbuildinfo` files that caches information about the last build, so that if you run `yarn build` and then `yarn build` again it can consult the cache to determine which files have already been built. This command clears those files.

---

We can skip the shell script by doing this, which IMO is far more readable:
```suggestion
    "build:clean": "yarn rimraf './**/*.tsbuildinfo' './**/dist'",
```

In skunkworks every workspace that has cleanup to do has a `clean` script which we invoke using `foreach`: https://github.com/MetaMask/snaps-skunkworks/blob/1ef459c8d180f1355e2da6534edc8952b452a3bc/package.json/#L28

`foreach` does not error if the script doesn't exist, which is great.

---

Okay, I will take a look at the pattern `snaps-skunkworks` is following (as it seems like y'all have already figured this out) and incorporate your suggestion here if necessary.

---

Simplified in [be2bbe1](https://github.com/MetaMask/controllers/pull/831/commits/be2bbe17c5a97d18b64ebc127f19875e86a9fee3).